ocean_lib/ocean/ocean_assets.py

```diff
@@ -473,32 +470,8 @@ def download_file(self, did: str, wallet) -> str:
         ddo = self.resolve(did)
         datatoken_address = ddo.datatokens[0]["address"]
         datatoken = Datatoken(self._config_dict, datatoken_address)
+        datatoken.dispense_if_free(wallet)
 
```

Comment:
I think the flow should be:
 - If I already have 1.0 DT in my wallet, go further
 - If not, call datatoken.dispense_if_free